Modules/Server/Sources/PocketCastsServer/Public/Sync/SyncYearListeningHistoryTask.swift

```diff
@@ -95,6 +95,9 @@ class SyncYearListeningHistoryTask: ApiBaseTask {
                 performRequest(yearToSync: year, token: token, shouldSync: true)
             } else {
                 success = true
+
+                // If there are no episodes to sync we leave the dispatch group
+                totalEpisodesDispatchGroup.leave()
```

Comment:
There's a way to improve this (unfortunate) collection of dispatch groups. However I think that for the release branch we should be careful so I'm doing as less changes as possible.